modules/lifecycle-management-impl/src/main/java/org/opencastproject/lifecyclemanagement/impl/PolicyCheckRunner.java

```diff
@@ -0,0 +1,284 @@
+  private User systemAdminUser;
+  private Organization defaultOrganization;
+
+  private Set<RepeatingPolicyRunner> repeatingPolicyRunners = new HashSet<>();
+
+  @Reference
+  public void setLifeCycleService(LifeCycleService lifeCycleService) {
+    this.lifeCycleService = lifeCycleService;
+  }
+
+  @Reference
+  public void setSecurityService(SecurityService securityService) {
+    this.securityService = securityService;
+  }
+
+  @Reference
+  public void setOrganizationDirectoryService(OrganizationDirectoryService orgDirServ) {
+    this.organizationDirectoryService = orgDirServ;
+  }
+
+  @Activate
+  public void activate(BundleContext ctx) {
+    logger.info("Activating LifeCycle Management Policy Checker.");
+    this.defaultOrganization = new DefaultOrganization();
+    String systemAdminUserName = ctx.getProperty(SecurityUtil.PROPERTY_KEY_SYS_USER);
+    this.systemAdminUser = SecurityUtil.createSystemUser(systemAdminUserName, defaultOrganization);
+
+    scheduledExecutor = Executors.newScheduledThreadPool(1);
+    scheduledExecutor.scheduleWithFixedDelay(new Runner(), 10, 5,
+        TimeUnit.SECONDS);
+  }
+
+  @Deactivate
+  public void deactivate() {
+    if (scheduledExecutor != null) {
+      try {
+        scheduledExecutor.shutdownNow();
+        if (!scheduledExecutor.isShutdown()) {
+          logger.info("Waiting for Dispatcher to terminate");
+          scheduledExecutor.awaitTermination(10, TimeUnit.SECONDS);
+        }
+      } catch (InterruptedException e) {
+        logger.error("Error shutting down the Dispatcher", e);
+      }
+    }
+  }
+
+  class Runner implements Runnable {
+    @Override
+    public void run() {
+      try {
+        logger.debug("PolicyCheck runs");
+
+        List<Organization> orgs = organizationDirectoryService.getOrganizations();
+
+        for (Organization org : orgs) {
+          SecurityUtil.runAs(securityService, org, systemAdminUser, () -> {
+
+            // Get all (active) policies
+            List<LifeCyclePolicy> policies;
+            try {
+              policies = lifeCycleService.getActiveLifeCyclePolicies();
+            } catch (NullPointerException e) {
+              logger.info("NPE: ", e);
+              return;
+            }
+
+            for (LifeCyclePolicy policy : policies) {
+              if (!policy.isActive()) {
+                logger.debug("Policy " + policy.getTitle() + " is deactivated. Skipping.");
+                continue;
+              }
+              // If isSpecificDate
+              if (policy.getTiming() == Timing.SPECIFIC_DATE) {
+                logger.debug("Policy " + policy.getTitle() + " with specific date " + policy.getActionDate());
+                // If current date > action date
+                if (policy.getActionDate().before(new Date())) {
+
+                  try {
+                    // Deactivate policy
+                    policy.setActive(false);
+                    lifeCycleService.updateLifeCyclePolicy(policy);
```

Comment:
Yes. So it would better to set the policy to "inactive" until after all tasks have been created?

Reply:
I think so yeah.  Can we track the previously created tasks so we can cancel them, or am I shaving a yak here?

Reply:
That should be possibly, I'm not quite on the where and when we would want to cancel the tasks though. In the case of a crash halfway through this, arguably there would be no need to cancel already created tasks? The policy would be run again (since it was not set to inactive), and already created tasks won't be created again.